<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>IGTminer</title>
  <link rel='icon' type='image/x-icon' href='icon.png'/>
  <meta http-equiv="Content-Type" content="text/html" />
  <link rel="stylesheet" href="style.css" type="text/css"/>
</head>
<body>
<div class="forepanel">
<!-- header -->
<div style="text-align:center;">
<img src="logo.png" style="height:80px"><br>
<p style="font-size:15pt;color:#696969;margin:10px">A tool for nucleus organelle gene map construction</p>
</div>
<!-- header-tabs -->
<div style="text-align:center;background-color:#F5F5F5">
  	<div id="header-tabs" class="header-tabs">
			<ul>
			  <li><a href="https://chenym1.github.io/IGTminer/"><span>Home</span></a></li>
			  <li><a href="https://github.com/chenym1/IGTminer"><span>Source Code</span></a></li>
			  <li><a href="https://chenym1.github.io/IGTminer/tutorial.html"><span>Tutorial</span></a></li>
			  <li><a href="https://chenym1.github.io/IGTminer/qa.html"><span>Q&A</span></a></li>
			  <li><a href="http://wheat.cau.edu.cn/pNOGmap/"><span><img src="NOG_logo.png" style="vertical-align:middle;height:20px">pNOGmap</span></a></li>
			</ul>
		</div>
</div>


<div class="info">

<h1>Questions and Answers</h1>

<div style="padding-bottom:5px">
<h3>Q: What are the main steps of IGTminer?</h3>
<p style="font-size:10pt">
  A: IGTminer takes multiple assemblies as input and produces a NOG map connnecting NOGs across assemblies. The main steps includes: <b>1)</b> Reannotation of NOGs in the whole genome; <b>2)</b> Identification of NOGCs; <b>3)</b></b> Improvement of pairwise collinear block search; <b>4)</b> Collinearity network construction.
</p>
</div>

<div style="padding-bottom:5px">
<h3>Q: What are the advantages of IGTminer?</h3>
<p style="font-size:10pt">
  1. Many genomes are independently annotated by diverse pipelines, which results low comparability across multiple assemblies. All organelle-to-nucleus gene transfers in the reference genome will be identified and reannotated with IGTminer.<br>
  2. A polyploid assembly will be decomposed into multiple diploid subassemblies to meet the need for phylogenomic analysis. To avoid issue with inaccuracy of collinear block search, IGTminer substitutes NOGCs into single sites to improve the search power of the collinear block.
</p>
</div>

</div>


<!--footer-->
<div style="margin-top:10px">
  <div style="background-color:#F5F5F5;height:5px"></div>
	<center style="margin-top:5px">
	 <table><tr>
    <td><a href="http://www.cau.edu.cn/"><img src="cau.png" style="vertical-align:middle;height:50px"></a></td>
    <td><div class="line" style="height:30px;width:2px;background-color:#8AB08D;vertical-align:middle;margin:5px"></div></td>
    <td><a href="http://wheat.cau.edu.cn/"><img src="wggc.png" style="vertical-align:middle;height:50px"></a></td>
    <td><div class="line" style="height:30px;width:2px;background-color:#8AB08D;vertical-align:middle;margin:5px"></div></td>
    <td><div class="line" style="height:50px;width:15px;"></div></td>
    <td style="width:50px; height:50px;">
<div style="vertical-align:middle;width:100%;height:100%;padding-top:10px"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/4.js?i=5x1743v7tv3&amp;m=0&amp;h=50&amp;c=ff0000&amp;r=0" async="async"></script></div>
    </td>
   </tr></table>
  </center>
  <center><p style="font-size:8pt;margin-top:5px;margin-bottom:4px">
      Copyright &copy; 2022-2025 All rights reserved<br>
      Technical contact: Chen, Yongming; E-mail: chen_yongming@126.com<br>
      Corresponding contact: Guo, Weilong; E-mail: guoweilong@cau.edu.cn
  </p></center>
</div>


</div>
</body>
</html>