<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>IGTminer</title>
  <link rel='icon' type='image/x-icon' href='icon.png'/>
  <meta http-equiv="Content-Type" content="text/html" />
  <link rel="stylesheet" href="style.css" type="text/css"/>
</head>
<body>
<div class="forepanel">
<!-- header -->
<div style="text-align:center;">
<img src="logo.png" style="height:80px"><br>
<p style="font-size:15pt;color:#696969;margin:10px">A tool for nucleus organelle gene map construction</p>
</div>
<!-- header-tabs -->
<div style="text-align:center;background-color:#F5F5F5">
  	<div id="header-tabs" class="header-tabs">
			<ul>
			  <li><a href="https://chenym1.github.io/IGTminer/"><span>Home</span></a></li>
			  <li><a href="https://github.com/chenym1/IGTminer"><span>Source Code</span></a></li>
			  <li><a href="https://chenym1.github.io/IGTminer/tutorial.html"><span>Tutorial</span></a></li>
			  <li><a href="https://chenym1.github.io/IGTminer/qa.html"><span>Q&A</span></a></li>
			  <li><a href="http://wheat.cau.edu.cn/pNOGmap/"><span><img src="NOG_logo.png" style="vertical-align:middle;height:20px">pNOGmap</span></a></li>
			</ul>
		</div>
</div>


<div class="info">

<h1>Tutorial</h1>
<div style="padding-bottom:5px">
  
<!--isntall-->
<h3 style="color:#008B8B">Install</h3>
<p style="font-size:12pt;background-color:#CFCFCF;padding:10px;width:500px">
git clone https://github.com/chenym1/IGTminer.git<br>
cd IGTminer<br>
IGTminer -h
</p>
<!--dependencies-->
<h4>Dependencies</h4>
<p style="font-size:10pt">
  <span style="font-size:12pt">IGTminer is implemented in python3 and CentOS Linux, and external command-line tools, which are executable on your PATH, are needed:</span><br>
<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?">BLAST</a> (v2.9.0)&emsp;&emsp;The installing way: <code>conda install blast -c bioconda</code><br>
<a href="https://github.com/tanghaibao/jcvi/">MCscan</a> (v1.0.6)&emsp;&emsp;The installing way: <code>pip install jcvi</code><br>
<a href="https://bedtools.readthedocs.io/en/latest/">BEDTools</a> (v2.29.2)&emsp;&emsp;The installing way: <code>conda install bedtools -c bioconda</code><br>
<a href="https://www.r-project.org/">R</a> (v3.6.0)
</p>


<!--Usage-->
<h3 style="color:#008B8B">Usages</h3>


<h4>IGTminer</h4>
<pre style="font-size:12pt;background-color:#CFCFCF;padding:10px;width:700px"><code style="background-color:#CFCFCF">$ IGTminer -h

Program:      IGTminer
Description:  A tool for nucleus organelle gene (NOG) map construction
Version:      v1.0.0

Usage:        IGTminer &#60;command&#62; [options]

Subcommands include:

[ pipeline ]
     reannotation          Reannotation of nucleus organelle genes (NOGs)
     collinearityNetwork   Pairwise collinear block search and construction of NOG map
</code></pre>

<h4>reannotation - Reannotation of nucleus organelle genes</h4>
<pre style="font-size:12pt;background-color:#CFCFCF;padding:10px;width:700px"><code style="background-color:#CFCFCF">$ IGTminer reannotation -h

Tool:         IGTminer reannotation
Description:  Reannotation of nucleus organelle genes (NOGs)

Usage:        IGTminer reannotation [options]

Options:
  -h          Show this message and exit
  -g &#60;file&#62;   Gff3
  -f &#60;file&#62;   Nucleus genome fasta
  -c &#60;file&#62;   Chromosome name file
  -r &#60;file&#62;   Custom protein sequence file
  -m &#60;str&#62;    If parameter '-r' is empty, 
               please select plastid (pt) or mitochondrion (mt) [default pt]
  -p &#60;file&#62;   Organelle genome fasta
  -d &#60;int&#62;    Maximum distance for clustering NOGs [default 140000]
  -n &#60;str&#62;    Prefix name of output file [default NOG]
  -t &#60;int&#62;    Number of threads [default 12]
</code></pre>

<h4>collinearityNetwork - Pairwise collinear block search and construction of NOG map</h4>
<pre style="font-size:12pt;background-color:#CFCFCF;padding:10px;width:700px"><code style="background-color:#CFCFCF">$ IGTminer collinearityNetwork -h

Tool:         IGTminer collinearityNetwork
Description:  Pairwise collinear block search and construction of NOG map

Usage:        IGTminer collinearityNetwork [options]

Options:
  -h          Show this message and exit
  -c &#60;file&#62;   Chromosome name path
  -n &#60;file&#62;   Prefix name of output file [default IGTminerOutput]
  -e          E-value of BLASTP [default 1e-5]
  -d &#60;int&#62;    Extent of flanking regions to search using MCscan [default 8]
  -s &#60;int&#62;    Minimum number of anchors in a cluster using MCscan [default 10]
  -t          Number of threads for blast [default 12]
</code></pre>

<!--example-->
<h3 style="color:#008B8B">Example</h3>
<pre style="font-size:12pt;background-color:#CFCFCF;padding:10px;width:700px"><code style="background-color:#CFCFCF">mkdir example; cd example
wget http://wheat.cau.edu.cn/TGT/downloads/example.tar.gz
tar -xzvf example.tar.gz
IGTminer reannotation -g te.gff3 -f te.fa -c te.chrlist -p te.pt.fa -n te
IGTminer reannotation -g svevo.gff3 -f svevo.fa -c svevo.chrlist -p svevo.pt.fa -n svevo
IGTminer collinearityNetwork -c chrlist
</code></pre>

</div>
</div>
<!--footer-->
<div style="margin-top:10px">
  <div style="background-color:#F5F5F5;height:5px"></div>
	<center style="margin-top:5px">
	 <table><tr>
    <td><a href="http://www.cau.edu.cn/"><img src="cau.png" style="vertical-align:middle;height:50px"></a></td>
    <td><div class="line" style="height:30px;width:2px;background-color:#8AB08D;vertical-align:middle;margin:5px"></div></td>
    <td><a href="http://wheat.cau.edu.cn/"><img src="wggc.png" style="vertical-align:middle;height:50px"></a></td>
    <td><div class="line" style="height:30px;width:2px;background-color:#8AB08D;vertical-align:middle;margin:5px"></div></td>
    <td><div class="line" style="height:50px;width:15px;"></div></td>
    <td style="width:50px; height:50px;">
<div style="vertical-align:middle;width:100%;height:100%;padding-top:10px"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/4.js?i=5x1743v7tv3&amp;m=0&amp;h=50&amp;c=ff0000&amp;r=0" async="async"></script></div>
    </td>
   </tr></table>
  </center>
  <center><p style="font-size:8pt;margin-top:5px;margin-bottom:4px">
      Copyright &copy; 2022-2024 All rights reserved<br>
      Technical contact: Chen, Yongming; E-mail: chen_yongming@126.com<br>
      Corresponding contact: Guo, Weilong; E-mail: guoweilong@cau.edu.cn
  </p></center>
</div>


</div>
</body>
</html>